cli/src/actions/locations.rs

```
@@ -233,118 +249,140 @@ fn yaml_to_property_values(
-                    .map_err(|err| CliError::PayloadError(format!("{}", err)))?;
-                property_values.push(property_value);
-            }
-            schemas::DataType::Number => {
-                let property_value = PropertyValueBuilder::new()
-                    .with_name(def.name.clone())
-                    .with_data_type(def.data_type.into())
-                    .with_number_value(serde_yaml::from_value(value.clone())?)
-                    .build()
-                    .map_err(|err| CliError::PayloadError(format!("{}", err)))?;
-                property_values.push(property_value);
-            }
-            schemas::DataType::String => {
-                let property_value = PropertyValueBuilder::new()
-                    .with_name(def.name.clone())
-                    .with_data_type(def.data_type.into())
-                    .with_string_value(serde_yaml::from_value(value.clone())?)
-                    .build()
-                    .map_err(|err| CliError::PayloadError(format!("{}", err)))?;
-                property_values.push(property_value);
+        match get_property_value_from_property_definition(&properties, def) {
+            Ok(Some(property_value)) => property_values.push(property_value),
+            Ok(None) => (),
+            Err(CliError::UserError(message)) | Err(CliError::PayloadError(message)) => {
+                error!("{}", message);
+                are_errors_present = true;
             }
-            schemas::DataType::Enum => {
-                let property_value = PropertyValueBuilder::new()
-                    .with_name(def.name.clone())
-                    .with_data_type(def.data_type.into())
-                    .with_enum_value(serde_yaml::from_value(value.clone())?)
-                    .build()
-                    .map_err(|err| CliError::PayloadError(format!("{}", err)))?;
-                property_values.push(property_value);
+            Err(CliError::EmptyError) => {
```

Because of the recursive nature of the function and its helper, we don't want to report the same error at multiple nested levels. We just want to note that an error occurred, and have confidence that the causes of EmptyError (Payload or User Errors) have been reported via the `error!` macro. an emptyError occurring at any level of nesting will eventually be handled with proper verbiage higher up on the call stack, as found on lines 204 and 233.